client/asset/doge/doge.go

```diff
@@ -163,32 +177,73 @@ func NewWallet(cfg *asset.WalletConfig, logger dex.Logger, network dex.Network)
 		SingularWallet:           true,
 		UnlockSpends:             true,
 		ConstantDustLimit:        dustLimit,
-		FeeEstimator: func(cl btc.RawRequester, _ uint64) (uint64, error) {
-			confArg, err := json.Marshal(feeConfs)
-			if err != nil {
-				return 0, err
-			}
-			resp, err := cl.RawRequest("estimatefee", []json.RawMessage{confArg})
-			if err != nil {
-				return 0, err
-			}
-			var feeRate float64
-			err = json.Unmarshal(resp, &feeRate)
-			if err != nil {
-				return 0, err
-			}
-			if feeRate <= 0 {
-				return 0, nil
-			}
-			// estimatefee is f#$%ed
-			// https://github.com/decred/dcrdex/pull/1558#discussion_r850061882
-			if feeRate > dexdoge.DefaultFeeRateLimit/1e5 {
-				return dexdoge.DefaultFee, nil
-			}
-			return uint64(math.Round(feeRate * 1e5)), nil
-		},
-		BlockDeserializer: dexdoge.DeserializeBlock,
+		FeeEstimator:             estimateFee,
+		ExternalFeeEstimator:     fetchExternalFee,
+		BlockDeserializer:        dexdoge.DeserializeBlock,
 	}
 
 	return btc.BTCCloneWallet(cloneCFG)
 }
+
+// NOTE: btc.(*baseWallet).feeRate calls the local and external fee estimators
+// in sequence, applying the limits configured in baseWallet.
```

Comment:
This is the gist of the design in this PR.  A wallet choose to set these functions or not.  If the local estimator is not set, case the BTC RPC default is used.  If the external estimator is not set, no external estimate is obtained.

I'm splitballing with the `NoLocalFeeRate` dummy idea for the local estimator if an instance needs to be designed that only does external estimates but there is no possibility for local estimates.  We can consider this for SPV, but we may not want to implement `asset.FeeRater` for SPV wallets since we've established pretty well to use the fee rates from server that we have cached or fetched on-demand.